### PR TITLE
Return Status page

### PR DESCRIPTION
**Pull request by rickreyhsig**
https://www.pivotaltracker.com/story/show/186522352

Force the URL to: http://statefile.localhost:3000/en/az/questions/return-status
As the submission confirmation page does not link to this.

| Submission Status  | Screenshot |
| ------------- | ------------- |
| **Pending** |  Screen Shot 2023-12-08 at 5 19 13 PM  |
| **Rejected**  |  Screen Shot 2023-12-08 at 5 18 46 PM  |
| **Accepted Owed**  |  Screen Shot 2023-12-08 at 6 00 45 PM  |
| **Accepted Refund**  |  Screen Shot 2023-12-08 at 5 49 28 PM  |

### CODE REVIEW

**Review by bengolder — APPROVED** [review]
Approved with non-blocking suggestions for improvement.

**Review by bengolder — COMMENTED** [review]
Approved with non-blocking suggestions for improvement.

File: app/models/state_file_base_intake.rb [hunk 1]
```diff
@@ -144,4 +144,15 @@ def save_nil_enums_with_unfilled
...
+  def return_status
+    case self.efile_submissions.last.current_state
+    when 'accepted'
+      'accepted'
+    when 'rejected'
+      'rejected'
+    else
+      'pending'
+    end
+  end
```
[the remark this answers is not in the document]
Comment (rickreyhsig) [reply app/models/state_file_base_intake.rb:157]: Done

File: spec/models/state_file_base_intake_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    end
+
+    context "with a rejected return" do
+      before { allow_any_instance_of(EfileSubmission).to receive(:current_state).and_return('rejected') }
+      it "it is rejected" do
+        expect(intake.return_status).to eq('rejected')
+      end
+    end
+
+    context "with a pending return" do
```
[the remark this answers is not in the document]
Comment (rickreyhsig) [reply spec/models/state_file_base_intake_spec.rb:24]: Done

File: spec/models/state_file_base_intake_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      before { allow_any_instance_of(EfileSubmission).to receive(:current_state).and_return('pending') }
```
[the remark this answers is not in the document]
Comment (rickreyhsig) [reply spec/models/state_file_base_intake_spec.rb:25]: Done

File: app/controllers/state_file/questions/return_status_controller.rb
[the remark this answers is not in the document]
Comment (rickreyhsig) [reply app/controllers/state_file/questions/return_status_controller.rb:1]: Done

**Pull request merged**